### PR TITLE
Reduced-order syngen models

### PR DESCRIPTION
**Pull request by dinkelbachjan**
This PR covers the implementation of iterative reduced-order synchronous generator models, namely the PCM and TPM models. Besides, it removes the DCIM implementations, which are covered now by PCM models with zero iterations in the DP domain. It also refactors the VBR model implementations and involves a new DP-DQ interface, which is used by all DP reduced-order synchronous generator models (PCM, TPM, VBR) and shall be reused by other DQ based components in the future. Furthermore, it adds several validation notebooks to compare the different reduced-order models against each other and to compare the results of the 4th order VBR model in the WSCC benchmark system against results from PSAT.

Based on #199 
Rebase of #207  
Related to #75, #221 and #225

### CODE REVIEW

File: dpsim-models/include/dpsim-models/DP/DP_Ph1_SynchronGenerator4OrderPCM.h [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		///
+		void mnaCompApplyRightSideVectorStamp(Matrix& rightVector) final;
+		///
+		void mnaCompPostStep(const Matrix& leftVector) final;
+		///
+		void mnaCompApplySystemMatrixStamp(SparseMatrixRow& systemMatrix) final {};
+
+	protected:
+		/// Interface used to transform between DP and DQ vars
+		DPDQInterface mDomainInterface;
```
Comment (martinmoraga) [inline dpsim-models/include/dpsim-models/DP/DP_Ph1_SynchronGenerator4OrderPCM.h:65]: This variable is already defined in the class ReducedOrderSynchronGeneratorVBR

**Comment by martinmoraga** [comment]
Do we really need the variable `mIter ` in [MNASolverDirect.h](https://github.com/sogno-platform/dpsim/pull/213/files#diff-292b7e0cdf57bf8af48543c1b511e8b67a6440f94182113595c4863518d7d8b2)?

**Comment by martinmoraga** [comment]
@dinkelbachjan Maybe we have to delete the model [SP_Ph1_SynchronGenerator4OrderDCIM.cpp](https://github.com/sogno-platform/dpsim/pull/213/files#diff-6b043e994f52845900ccc5ddae40feabe84ef8fbda641b4b966db595995b4a58). It does not work...

**Comment by martinmoraga** [comment]
@dinkelbachjan Do we really need the variable `mMaxIterations` in [dpsim/include/dpsim/Solver.h](https://github.com/sogno-platform/dpsim/pull/213/files#diff-203f46a4c9c7f7a6050f0b0ddff7f38b7f0e142990987a7cd7400c60b1be1171)

File: dpsim/include/dpsim/MNASolverDirect.h [hunk 2]
```diff
@@ -167,6 +168,9 @@ namespace DPsim {
 		/// log LU decomposition times
 		void logLUTimes() override;
 
+		/// ### SynGen Interface ###
+		int mIter = 0;
```
Comment (dinkelbachjan) [inline dpsim/include/dpsim/MNASolverDirect.h:172]: Do we really need this variable?

**Review by martinmoraga — APPROVED** [review]

File: dpsim/include/dpsim/MNASolverDirect.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (martinmoraga) [reply dpsim/include/dpsim/MNASolverDirect.h:172]: Issue #225

[the remark this answers is not in the document]
Comment (dinkelbachjan) [reply dpsim/include/dpsim/MNASolverDirect.h:172]: We keep the variable `mIter` for now, as we intend to make use of it, when taking care of #225.

File: dpsim-models/src/SP/SP_Ph1_SynchronGenerator4OrderDCIM.cpp
[the remark this answers is not in the document]
Comment (dinkelbachjan) [reply dpsim-models/src/SP/SP_Ph1_SynchronGenerator4OrderDCIM.cpp:1]: Yes, I will remove the model. If we want to have the model available again in the future, we can rather implement it analagously to `DP_Ph1_SynchronGenerator4OrderPCM`.

**Review by martinmoraga — APPROVED** [review]

**Pull request merged**